core/src/main/scala/medeia/generic/CoproductEncoderInstances.scala

```diff
@@ -7,15 +7,16 @@ import shapeless.labelled.FieldType
 import shapeless.{:+:, CNil, Coproduct, Inl, Inr, Witness}
 
 trait CoproductEncoderInstances {
-  implicit def cnilEncoder[Base]: ShapelessEncoder[Base, CNil] = _ => throw new Exception("Inconceivable!")
+  implicit def cnilEncoder[Base]: ShapelessEncoder[Base, CNil] = (_, _) => throw new Exception("Inconceivable!")
   implicit def coproductEncoder[Base, K <: Symbol, H, T <: Coproduct](
       implicit
       witness: Witness.Aux[K],
       hEncoder: Lazy[BsonDocumentEncoder[H]],
       tEncoder: ShapelessEncoder[Base, T],
       options: SealedTraitDerivationOptions[Base] = SealedTraitDerivationOptions[Base]()
   ): ShapelessEncoder[Base, FieldType[K, H] :+: T] = {
-    case Inl(head) => hEncoder.value.encode(head).append(options.discriminatorKey, BsonString(options.transformDiscriminator(witness.value.name)))
-    case Inr(tail) => tEncoder.encode(tail)
+    case (Inl(head), _) =>
```

Comment:
I know that ignoring the accumulator doc in CoproductEncoderInstances is not nice. But as CoproductEncoder uses BsonDocumentEncoder as hEncoder to not be forced to use Shapeless for the children of the sealed trait hierarchy I don't see a better way currently. Do you?

Reply:
does that only work because we `mutate` :scream: the accumulator in other cases? :see_no_evil:

Reply:
nope, for coproduct we don't need an accumulator as it does not accumulate things, as it is either one or the other, so no accumulating recursion

Reply:
Okay makes sense :shrug: